include/envoy/http/header_map.h

```diff
@@ -86,6 +86,16 @@ class HeaderString {
    */
   const char* c_str() const { return buffer_.ref_; }
 
+  /**
+   * @return a std::string.
+   */
+  std::string getString() const {
```

Comment:
Is this really necessary? Seems pretty superfluous to me, but I don't feel that strongly about it. If it is, I suspect you don't need the if statement and the constructor will do the right thing if you pass a length of 0, but I'm not sure.

Reply:
the need for it came up in the feedback for pr https://github.com/envoyproxy/envoy/pull/2415
it is kind of useful to have it here and not have the user have to convert a `c_str()`.